### DISCUSSION TITLE
Wrangling dependencies

### DISCUSSION THREAD

**Discussion by GaryElshaw** [transmission/transmission#4090] [Ideas]
@ckerr @mikedld I was just wondering if this might be helpful for managing the various Transmission dependencies? https://github.com/renovatebot/renovate

**Comment by nevack** [comment]
Hi, Gary.
I've some experience with renovate.
I think it's good for keeping NPM or Maven dependencies up to date, I do not think it would play well with C++ deps.

In my opinion, it would be great to have some tool checking current dependencies for known vulnerabilities and suggest updating them with known good versions.
There's no good reason to update everything asap.

**Comment by ckerr** [comment]
I'm neutral on this.

I have no background with renovate. Just some cursory Googling [indicates](https://joht.github.io/johtizen/automation/2022/08/03/keep-your-cpp-dependencies-up-to-date.html) that using it to manage C++ deps is possible, though IDK how difficult it is.

My main concern is be that we don't automatically bump the **minimum** versions of dependencies. All other things being equal, it's better to use system-installed packages (i.e., `third-party/` should be the backup plan). So, no semver-major updates.

:+1: about the idea of checking known vulnerabilities.

For both suggestions: If someone wants to put in the work I'd be fine with it in principle & happy to review it.
